x-pack/legacy/plugins/snapshot_restore/plugin.ts

```diff
@@ -8,10 +8,10 @@ import { registerRoutes } from './server/routes/api/register_routes';
 import { Core, Plugins } from './shim';
 
 export class Plugin {
-  public start(core: Core, plugins: Plugins): void {
+  public start(core: Core, plugins: Plugins, isSlmEnabled: boolean): void {
```

Comment:
Suggestion: Instead of change this `Plugin.start`'s signature, perhaps we can add it to the `Plugins` interface defined in `snapshot_restore/shim.ts`? Maybe under it's own top level similar to what `cloud` entry has at the moment. What do you think?